controllers/databasecluster_controller.go

```diff
@@ -1855,7 +2015,7 @@ func (r *DatabaseClusterReconciler) reconcilePGBackupsSpec(
 	return newBackups, nil
 }
 
-func (r *DatabaseClusterReconciler) genPGDataSourceSpec(ctx context.Context, database *everestv1alpha1.DatabaseCluster) (*crunchyv1beta1.DataSource, error) {
+func (r *DatabaseClusterReconciler) genPGDataSourceSpec(ctx context.Context, database *everestv1alpha1.DatabaseCluster) (*crunchyv1beta1.DataSource, error) { //nolint:gocognit,lll,maintidx
```

Comment:
Can we split this function into smaller chunks instead of adding nolint for cognitive complexity, long line, and maintenance index?

Reply:
I observed we usually add `nolint` directives when it comes complexity or length of functions.
Other methods in the operator use `nolint` as well so I understand the linters serve more as a recommendation than a practice.
Do you think this practice is something we shall address in a more rigorous fashion?

Reply:
We can move some parts of this function to different functions to make the logical split of the function

Reply:
Absolutely.
If you feel we shall approach this in a more systematic way, let's open a new ticket and address this in a separate PR.

Since this approach is consistent with the rest of the codebase and does not make the codebase worse, I think it's fine and we can address the removal of selected `nolint` rules separately, if needed.

Reply:
@gen1us2k in this case, please fix it as well :) we need to have a general rule for our whole code

https://github.com/percona/percona-everest-backend/pull/158/files#diff-4485a00dd63af8089d0eda57e3745c25f0da2e1d2efe9f9038a2d93f9bf9559fR150
https://github.com/percona/percona-everest-backend/pull/158/files#diff-fa107c8ea97a323473ac245ac9066b063fde00bc9c445d3b1740303dda5901b2R140

Reply:
@hors https://jira.percona.com/browse/EVEREST-482 on it. Agree